components/director/internal/domain/apptemplateversion/repository.go

```diff
@@ -0,0 +1,102 @@
+package apptemplateversion
+
+import (
+	"context"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/resource"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/internal/repo"
+)
+
+const tableName string = `public.app_template_versions`
+
+var (
+	idTableColumns        = []string{"id"}
+	updatableTableColumns = []string{"title", "correlation_ids"}
+	tableColumns          = append(idTableColumns, []string{"version", "title", "correlation_ids", "release_date", "created_at", "app_template_id"}...)
+)
+
+// EntityConverter converts between internal model and entity
+//
+//go:generate mockery --name=EntityConverter --output=automock --outpkg=automock --case=underscore --disable-version-string
+type EntityConverter interface {
+	ToEntity(in *model.ApplicationTemplateVersion) *Entity
+	FromEntity(entity *Entity) *model.ApplicationTemplateVersion
+}
+
+type repository struct {
+	creator            repo.CreatorGlobal
+	existQuerierGlobal repo.ExistQuerierGlobal
+	singleGetterGlobal repo.SingleGetterGlobal
+	updaterGlobal      repo.UpdaterGlobal
+	listerGlobal       repo.ListerGlobal
+	conv               EntityConverter
+}
+
+// NewRepository returns a new entity responsible for repo-layer ApplicationTemplateVersion operations.
+func NewRepository(conv EntityConverter) *repository {
+	return &repository{
+		creator:            repo.NewCreatorGlobal(resource.Application, tableName, tableColumns),
+		existQuerierGlobal: repo.NewExistQuerierGlobal(resource.ApplicationTemplateVersion, tableName),
+		singleGetterGlobal: repo.NewSingleGetterGlobal(resource.ApplicationTemplateVersion, tableName, tableColumns),
+		updaterGlobal:      repo.NewUpdaterGlobal(resource.ApplicationTemplateVersion, tableName, updatableTableColumns, idTableColumns),
+		listerGlobal:       repo.NewListerGlobal(resource.ApplicationTemplateVersion, tableName, tableColumns),
+		conv:               conv,
+	}
+}
+
+// Create persists a model.ApplicationTemplateVersion
+func (r *repository) Create(ctx context.Context, item model.ApplicationTemplateVersion) error {
+	log.C(ctx).Debugf("Converting Application Template Version with id %s to entity", item.ID)
```

Comment:
maybe we first need to check if `item` is not nil?

Reply:
it is not a pointer so it won't be null at any time. The null check is in the svc layer